src/pybamm/dispatch/entry_points.py

```diff
@@ -109,7 +119,35 @@ def __getattribute__(self, name):
 models = EntryPoint(group="pybamm_models")
 
 
-def Model(model: str, *args, **kwargs):
+def _get_cache_dir() -> Path:
+    cache_dir = Path(user_cache_dir(APP_NAME, APP_AUTHOR)) / "models"
+    cache_dir.mkdir(parents=True, exist_ok=True)
+    return cache_dir
+
+
+def get_cache_path(url: str) -> Path:
+    cache_dir = _get_cache_dir()
+    file_hash = hashlib.md5(url.encode()).hexdigest()
+    return cache_dir / f"{file_hash}.json"
+
+
+def clear_model_cache() -> None:
+    cache_dir = _get_cache_dir()
+    for file in cache_dir.glob("*.json"):
+        try:
+            file.unlink()
+        except Exception as e:
```

Comment:
Note, we also store some of PyBaMM's data files in the cache dir using a [`pooch`](https://github.com/fatiando/pooch) registry: https://github.com/pybamm-team/PyBaMM/blob/a1aa02c2ca108816b69294ee0287cf6729308424/src/pybamm/pybamm_data.py#L126-L139

I wonder if we could reuse some of the code here, because clearing the cache directory of JSON files could have unintended side effects. We do have JSON files there: https://github.com/pybamm-team/pybamm-data/releases/tag/v1.0.1

Could we perhaps rewrite the PR to use `pooch` instead? That will also provide safer defaults than using `urllib.request.urlretrieve()` directly, and we could rely on it for the checksum/caching bits to see if we need to download the model JSON again or not.